src/main/java/org/example/hospital/convertors/DepartmentConverter.java

```diff
@@ -0,0 +1,26 @@
+package org.example.hospital.convertors;
+
+import org.example.hospital.dto.DepartmentDTO;
+import org.example.hospital.entity.Department;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DepartmentConverter implements Converter<Department, DepartmentDTO> {
+    @Override
+    public DepartmentDTO convertToDTO(Department entity, DepartmentDTO dto) {
+        dto.setDepartmentId(entity.getDepartmentId());
+        dto.setName(entity.getName());
+        dto.setDescription(entity.getDescription());
+        return dto;
+    }
+
+    @Override
+    public Department convertToEntity(DepartmentDTO dto, Department entity) {
+        if (entity.getDepartmentId() == null) {
```

Comment:
I guess there is no need to check Id is exists or not here, you do it in Service layer

Reply:
There is no need to set an Id during an update (Id will be overwritten and cause an error), so I have a null checker to set Id only for new entities.